api/cms.go

```diff
@@ -149,3 +164,34 @@ func concludeRequestFile(c *gin.Context, fileName string, err *HttpError) {
 		err.Abort(c)
 	}
 }
+
+func handleItemsRequest(filePattern string, language string) ([]string, *HttpError) {
+	for _, lang := range consts.I18N_LANG_ORDER[language] {
+		pattern := fmt.Sprintf(filePattern, lang)
+		if files, err := filepath.Glob(pattern); err == nil {
+			return files, nil
+		}
+	}
+
+	return nil, NewNotFoundError()
+}
+
+// responds with File(s) content or aborts the request with the given error.
+func concludeRequestFiles(c *gin.Context, fileNames []string, err *HttpError) {
+	if err != nil || len(fileNames) == 0 {
+		err.Abort(c)
+	}
+
+	content := make([]string, len(fileNames))
+	var (
+		data []byte
+		errr error
+	)
+	for _, file := range fileNames {
+		if data, errr = os.ReadFile(file); errr != nil {
+			c.String(http.StatusOK, "[]")
```

Comment:
This doesn't make sense to me.
Upon errors, you want to respond with 200 ok ? I'd expect some 40x or 50x error status code.
Anyway, you should return after this line or otherwise you'll override the response in line 196

Reply:
changed to Abort